components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -300,6 +307,12 @@ abstract class AbstractFetchDownloadService : Service() {
         clearAllDownloadsNotificationsAndJobs()
     }
 
+    // As we are not fulfilling the api contract of a foreground service, we can be easily killed
+    // by the system when low on memory, without any call to onDestroy().
+    // To fulfill the contract, we need a pinned notification and call startForeground().
+    // This means we can left stuck notifications.
```

Comment:
nit: can be left with stuck notifications? can have leftover notifications?

Reply:
Thanks, let's go with "can be left with stuck notifications"